test/Chainweb/Test/Pact/DbCacheTest.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+module Chainweb.Test.Pact.DbCacheTest (tests) where
+
+import Chainweb.Pact.Backend.DbCache
+
+import Control.Monad.State.Strict
+
+import Data.Aeson
+import Data.ByteString.Lazy (toStrict)
+
+import Database.SQLite3.Direct
+
+import GHC.Compact
+
+import Test.Tasty
+import Test.Tasty.HUnit
+
+tests :: TestTree
+tests = testGroup "Chainweb.Test.Pact.DbCacheTest"
+    [ testCache ]
+
+-- | Create entries with a size the corresponds to the default
+-- chunk size of the cache.
+--
+entry :: MonadIO m => String -> m ([String], Int)
+entry c = do
+    s <- liftIO $ compactSize =<< compact [c]
```

Comment:
Can we use `compactCacheEntry` here? It would get rid of the `ghc-compact` dep and not have us replicating compaction code in the test.

Reply:
well, it may pay off if we refactor the tests to move the encode step out of `doCheck` below.

Reply:
Hmm, I think, it may not be the best idea, because the test wouldn't cover creation of the compact region any more, because we would give it already a compact region as input. The test would still work, but it would reduce coverage.